newrelic/core/application.py

```diff
@@ -1671,6 +1700,18 @@ def internal_agent_shutdown(self, restart=False):
 
         self.stop_data_samplers()
 
+        # Finishes collecting environment plugin information
+        # if this has not been completed during harvest
+        # lifetime of the application
+
+        while self.configuration.package_reporting.enabled and self._env_send:
```

Comment:
Just trying to match the logic above:
```suggestion
        while self.configuration and self.configuration.package_reporting.enabled and self._env_send:
```